### DISCUSSION TITLE
Cannot send webhook from AMS to a server which is using self-signed SSL certificate

### DISCUSSION THREAD

**Discussion by yashtandon113** [ant-media/Ant-Media-Server#6022] [FAQs] [answered]
When sending a https webhook from AMS to a server that is using a self-signed SSL certificate, the connection fails due to Tomcat not trusting the certificate. Is their any solution for this?

**Comment by lastpeony** [comment, marked as answer]
You need to create a truststore add your certificate file to it and then pass the truststore with its password as JVM argument to antmedia service for tomcat to trust the certificate and send succesfull webhook.
Here are the steps:
1-) Download and save your certificate file as yourCertificate.crt to the AMS server home directory. For this you can use wget or directly copy from your local with scp
2-) Run this command
`keytool -importcert -file yourCertificate.crt -keystore truststore.jks -alias someAlias`
You dont need to edit alias.
This will start a process of creating truststore.jks file. Set your password and create the file. 
3-) Go to `/etc/systemd/system/` directory.
Open antmedia.service with your favourite editor.
Add this to JVM arguments:
`-Djavax.net.ssl.trustStore=/home/truststore.jks -Djavax.net.ssl.trustStorePassword=yourTrustStorePassword`

![image](https://github.com/ant-media/Ant-Media-Server/assets/24750845/460f7e36-97e2-4096-ae7e-03a728dd589e)
Save and quit
4-)Run
`systemctl daemon-reload`
Run
`service antmedia restart`

Now AMS will trust self signed certificate and will be able to send webhooks to your server.
